synapse/handlers/saml_handler.py

```diff
@@ -294,6 +304,21 @@ def expire_sessions(self):
             del self._outstanding_requests_dict[reqid]
 
 
+def _check_attribute_requirement(ava: dict, req: SamlAttributeRequirement):
+    values = ava.get(req.attribute, [])
+    for v in values:
+        if v == req.value:
```

Comment:
Should we be ignoring whitespace here? (Or maybe the SAML2 code already strips all that out?)

Reply:
I'm not entirely sure we *should* ignore whitespace, but yes pysaml2 strips leading and trailing whitespace.